ui/app/components/job-status/panel/deploying.js

```diff
@@ -104,6 +104,14 @@ export default class JobStatusPanelDeployingComponent extends Component {
       let canary = alloc.isCanary ? 'canary' : 'nonCanary';
 
       if (allocationCategories[status]) {
+        // If status is failed or lost, we only want to show it IF it's used up its restarts/rescheds.
+        // Otherwise, we'd be showing an alloc that had been replaced.
+        if (alloc.willNotRestart) {
+          if (!alloc.willNotReschedule) {
+            // Dont count it
+            continue;
```

Comment:
Style thing: we don't use `continue` very often for breaking out of an iteration! It seemed apt here, though.